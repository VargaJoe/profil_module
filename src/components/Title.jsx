import React, { Component } from 'react';

class Title extends Component {
	constructor(props) {
		super(props);
		this.state = {};
	}

	render () {
		return (
			<h2 className="sn_title">
				{this.props.name}
			</h2>
		)
	}
};

export default Title;
