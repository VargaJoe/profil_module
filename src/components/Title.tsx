import * as React				from 'react';

class Title extends React.Component<any, any> {

	render () {
		return (
			<h2 className="sn_title">
				{this.props.name}
			</h2>
		);
	}
}

export default Title;
