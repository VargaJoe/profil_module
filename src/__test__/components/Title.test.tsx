import * as React 						  	from 'react';
import Title 							    from '../../components/Title';
import { 
	configure, 
		shallow,
		mount } 							from 'enzyme';
import * as Adapter 					  	from 'enzyme-adapter-react-16';
import toJson 				      			from 'enzyme-to-json';

configure( {adapter: new Adapter()} );

describe('<NoMatch /> shallow rendering', () => {
		
	const title = mount(<Title />); 
		
	// test Snapshot 
	it('Match to snapshot', () => {
		expect(toJson(title)).toMatchSnapshot();
	});

	// test Snapshot 
	it('Have <h2> tag', () => {
		let tagH2 = title.find('h2');
		expect(tagH2.length).toBe(1);
	});
		
});
