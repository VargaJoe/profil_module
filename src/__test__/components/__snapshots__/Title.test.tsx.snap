// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoMatch /> shallow rendering Match to snapshot 1`] = `
<h2
  className="sn_title"
/>
`;
